core/src/avm1/runtime.rs

```diff
@@ -27,6 +28,8 @@ pub struct Avm1<'gc> {
     /// don't close over the constant pool they were defined with.
     constant_pool: Gc<'gc, Vec<Value<'gc>>>,
 
+    constant_pool_cache: HashMap<SwfPosition, Gc<'gc, Vec<Value<'gc>>>>,
```

Comment:
If SWFs are repeatedly loaded then unloaded, this will cause a memory leak; this should probably be a `weak_table::WeakKeyHashMap` instead.